packages/react/src/CartLineQuantity.tsx

```diff
@@ -0,0 +1,32 @@
+import type {ComponentPropsWithoutRef, ElementType} from 'react';
+import {useCartLine} from './CartLineProvider.js';
+
+interface CartLineQuantityBaseProps<
+  ComponentGeneric extends ElementType = 'span'
+> {
+  /** An HTML tag or React Component to be rendered as the base element wrapper. The default is `span`. */
+  as?: ComponentGeneric;
+}
+
+export type CartLineQuantityProps<ComponentGeneric extends ElementType> =
```

Comment:
I need to level up my TS. As simple as this component is, the typings make it look scary.